### PR TITLE
Return response envelopes by value

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
We need to close on if response envelopes should return the raw response when available or not.  At present some code paths do include it which is why there are some test failures.

### CODE REVIEW

File: src/generator/operations.ts [hunk 1]
```diff
@@ -652,11 +683,12 @@ function generateResponseUnmarshaller(op: Operation, response: Response, imports
     if ((mediaType === 'XML' && schemaResponse.schema.type === SchemaType.Array) || schemaResponse.schema.language.go!.discriminatorInterface) {
       target = 'result';
     }
-    unmarshallerText += `\treturn &result, resp.UnmarshalAs${getMediaFormat(response.schema, mediaType, `&${target}`)}\n`;
+    unmarshallerText += `\terr := resp.UnmarshalAs${getMediaFormat(response.schema, mediaType, `&${target}`)}\n`;
```
Comment (catalinaperalta) [inline src/generator/operations.ts:686]: Why does this need to be separated now?

Comment (jhendrixMSFT) [reply src/generator/operations.ts:686]: It broke the case for multiple responses, where the return type is `interface{}`.  Might be a bug in Go, I didn't investigate further.

File: src/generator/operations.ts [hunk 2]
```diff
@@ -203,6 +203,36 @@ function generateMultiRespComment(op: Operation): string {
...
+function getZeroReturnValue(op: Operation, apiType: 'api' | 'op'): string {
+  if (!op.responses) {
+    // no responses return *http.Response
+    return 'nil';
+  }
+  let returnType = 'nil';
+  if (isLROOperation(op)) {
+    if (apiType === 'op') {
+      // the operation returns an *azcore.Response
+      returnType = 'nil';
```
Comment (catalinaperalta) [inline src/generator/operations.ts:215]: Just curious, when does this case happen?

File: src/generator/operations.ts (hunk 1 above)
Comment (catalinaperalta) [reply src/generator/operations.ts:686]: Ok good to know 👍

**Review by catalinaperalta — APPROVED** [review]
LGTM

File: src/generator/operations.ts (hunk 2 above)
Comment (jhendrixMSFT) [reply src/generator/operations.ts:215]: For operations that don't return a model (the `httpinfrastructuregroup` has a lot of examples).

Comment (catalinaperalta) [reply src/generator/operations.ts:215]: Ok I didnt see one with an LRO, but seems like a good idea just in case 👍

**Pull request merged**